### PR TITLE
Add a try_remove method

### PR DESCRIPTION
**Pull request by aym-v**
This PR adds a `try_remove` method that returns an Option on remove operations. It is [necessary](https://github.com/tokio-rs/tokio/issues/3558) to add a tokio_util::time::DelayQueue::try_remove method.

### CODE REVIEW

File: tests/slab.rs [hunk 1]
```diff
@@ -659,3 +659,14 @@ fn drain_rev() {
...
+
+#[test]
+fn try_remove() {
+    let mut slab = Slab::new();
+
+    let key = slab.insert(1);
+
+    assert_eq!(slab.try_remove(key), Some(1));
+    assert_eq!(slab.try_remove(key), None);
+    assert!(slab.get(key).is_none());
```
Comment (qm3ster) [inline tests/slab.rs:671]: Is there any narrative intent behind using `Eq` in one assert, and `.is_none()` in the other, vs using either one consistently throughout?

**Comment by qm3ster** [comment]
That is one nice looking git diff!

File: tests/slab.rs (hunk 1 above)
Comment (aym-v) [reply tests/slab.rs]: Thank you for pointing this out. The PR is updated

**Comment by taiki-e** [comment]
See also https://github.com/rust-lang/rust/pull/77480

**Review by qm3ster — APPROVED** [review]

**Comment by qm3ster** [comment]
Disposition to merge?

**Comment by zeenix** [comment]
> Disposition to merge?

@taiki-e ? Apart from other general benefits from this, we can also get rid of the [TODO here](https://github.com/smol-rs/async-executor/blob/master/src/lib.rs#L138) from a project we both maintain these days. :)

**Comment by taiki-e** [comment]
Sorry for the late review! Personally, for slab, I tend to prefer to return `Option` on remove like map or arena, rather than panic on remove like `Vec`. That said, changing `remove` is a breaking change, and we want to avoid major version bumps. Therefore, I think this is actually a reasonable solution...

**Review by taiki-e — APPROVED** [review]
Thanks @aym-v!

**Comment by taiki-e** [comment]
Published in v0.4.4.

**Pull request merged**